content/articles/jaspersoft-reports/index.md

```diff
@@ -150,72 +152,68 @@ import java.time.LocalDate;
 import java.util.List;
 
 public interface ReportService {
-    String generateReport(LocalDate localDate, String fileFormat) throws JRException, IOException;
-    List<Product> findAllProducts();
+ String generateReport(LocalDate localDate, String fileFormat) throws JRException, IOException;
+ List<Product> findAllProducts();
 }
 
 ```
-
 ### Design a report using Jaspersoft template
 
-First, open JasperSoft studio. Before we create a design of the report, we will add the MySQL connector to the classpath to ensure we can connect to our database and generate the desired fields from our table automatically.
-
-In Jaspersoft studio, click on *help* (located on the toolbar), *install new software*, *manage*, *java*, *build path*, *classpath variables*, *new* and add the jar file on the window that opens with the desired name as shown below:
+First, open JasperSoft studio. Before we create a design of the report, we will add the MySQL connector to the classpath. This will ensure we can connect to our database and generate the desired fields from our table. In Jaspersoft studio. Click on *help*, *install new software*, *manage*, *java*, *build path*, *classpath variables*,  *new*. Add the jar file on the window that opens with the desired name as shown below:
 
-![java connector](/jaspersoft-reports/jaspersoft-connector.jpg)
+![java connector](jaspersoft-connector.jpg)
 
-Once we have added the MySQL connector, we then run our spring boot app to generate the tables required by Jaspersoft studio.
+Add the MySQL connector and run the spring boot app.  The spring boot app will generate the tables required by Jaspersoft studio. Create a design for our report from existing templates. On the Jaspersoft toolbar click on *file*, *new*, *jasper report*. A window opens with a list of templates to choose from. For this tutorial, select the template named *simple blue* and click next. Add a new `.jrxml` file using the window that opens up. This is an extension for j
+Jasper report files. For our case name the file as `products.jrxml` and click next.
 
-To create a design of our report from existing templates, on the Jaspersoft toolbar click on *file*, *new*, *jasper report*, and a window opens with a list of templates to choose from. For this tutorial, select the template named *simple blue* and click next. Add a new `.jrxml` file using the window that opens up. This is simply an extension for jasper report files. For our case name the file as `products.jrxml` and click next.
+The window that opens next requires us to provide our data source. The data source is our database URL, username, and password to connect to our database. Click on new and on the data adapters window select database JDBC connection and click next. A window then opens where we need to fill the database connection properties as shown below.
```

Comment:
I was having problems here adding the data source based on your instructions. I had to open the driver classpath menu and manually add the MySQL connector jar I have saved on my computer. Perhaps there was an error in your instructions above for adding the MySQL classpath variables but I'm not sure what it is. Please have a look at this because it gave me a bit of a problem doing this and it might be for other readers.